### DISCUSSION TITLE
Video’s not playing in safari iOS browser (18.01)

### DISCUSSION THREAD

**Discussion by seanob86** [immich-app/immich#13223] [Q&A] [answered]
### The bug

Opening a video in iOS safari is not playing videos. Just a spinning circle.
Plays fine on desktop browser i.e. Firefox.
iOS apps on iPhone and iPad also play fine.

Also in the timeline and hovering over a video, that preview play back fine in browser
Same issue if trying to play back a live video.

No errors in the logs (via `docker logs -f immich`)
immich is behind Traefik reverse proxy
iOS version 18.01

### The OS that Immich Server is running on

Docker

### Version of Immich Server

v1.117.0

### Version of Immich Mobile App

v1.117.0

### Platform with the issue

- [ ] Server
- [ ] Web
- [ ] Mobile

### Your docker-compose.yml content

```YAML
Using unRaid imagegenius/immich image
```

### Your .env content

```Shell
Using unRaid imagegenius/immich image
```

### Reproduction steps

1. Access immich via web browser
2. Find a video and select it to open and play
3. Video does not play. Spinning circle appears over the play button (centre of video)
4. Attempt on other videos. Same behaviour

![IMG_1505](https://github.com/user-attachments/assets/50fb1d20-5920-47be-92e3-7542e45e7790)

**Comment by bo0tzz** [comment]
Have you changed any of the transcoding settings?

**Comment by seanob86** [comment]
I did enable `hardware decoding`, that's the only setting I changed before realizing that video playback is only failing on Safari browser on iPhone/iPad. Once I noticed that I did reverse and disabled `hardware decoding`, but there was no change, video playback still fail.

Video playback was working in safari previously but I don't know when it started having issues. i.e. could have started weeks ago without me realizing. Or it was due to the following changes this morning 
1. Enabling `hardware decoding`
2. iOS update (18.01, or even 18.0 a couple of weeks ago); or
3. Immich v.1.117 update (or one of the previous updates).

As I mentioned, using a browser on desktop PC (i.e. Firefox), there are zero issues

**Comment by seanob86** [comment]
Adding exported config json relevant transcoding config:

```
  "ffmpeg": {
    "crf": 23,
    "threads": 0,
    "preset": "medium",
    "targetVideoCodec": "h264",
    "acceptedVideoCodecs": [
      "h264"
    ],
    "targetAudioCodec": "aac",
    "acceptedAudioCodecs": [
      "aac",
      "mp3",
      "libopus"
    ],
    "acceptedContainers": [
      "mov",
      "ogg",
      "webm"
    ],
    "targetResolution": "720",
    "maxBitrate": "0",
    "bframes": -1,
    "refs": 0,
    "gopSize": 0,
    "npl": 0,
    "temporalAQ": false,
    "cqMode": "auto",
    "twoPass": false,
    "preferredHwDevice": "auto",
    "transcode": "required",
    "tonemap": "hable",
    "accel": "qsv",
    "accelDecode": false
  },
```

**Comment by dvbthien** [comment]
Hi @seanob86,
I think I have the same issue. Do you use a reverse proxy?

**Comment by seanob86** [comment]
> Hi @seanob86, I think I have the same issue. Do you use a reverse proxy?

Yeah I do. My immich instance is behind Traefik.

**Comment by dvbthien** [comment]
> Yeah I do. My immich instance is behind Traefik.

I use Traefik too. I think the issue is caused by Traefik, because when I don't use Traefik, the issue disappears. I still don't know how to fix it.

**Reply by seanob86** [reply]
> > Yeah I do. My immich instance is behind Traefik.
> 
> I use Traefik too. I think the issue is caused by Traefik, because when I don't use Traefik, the issue disappears. I still don't know how to fix it.

Weird. So i just tried playing a video from safari browser on my iPhone, and playback is working now. However on some occasions it doesn’t intermittently. On the iPad, no luck whatsoever. All while behind traefik reverse proxy.

I’ll have to try bypassing proxy by going directly to the ip when I have some time.

**Reply by dvbthien** [reply]
Yeah so weird, I can play the first video, but I can't play the second video. However, when I connect directly to the IP address, it works normally. When you have time, try connecting directly to the IP address. Let me know if you don't have issue when you connect directly to the IP address.

Seems like same the issue as: https://github.com/immich-app/immich/issues/10814

**Reply by seanob86** [reply]
> Yeah so weird, I can play the first video, but I can't play the second video. However, when I connect directly to the IP address, it works normally. When you have time, try connecting directly to the IP address. Let me know if you don't have issue when you connect directly to the IP address.
> 
> Seems like same the issue as: #10814

I just tried going directly to to the IP, effectively bypassing any reverse proxy. Still not working on the iPad Safari browser.
That issue you referenced appears to be the same that I am experiencing however on iPadOS Safari web browser. 

iPhone iOS Safari browser continues to be working without issue.

**Comment by seanob86** [comment, marked as answer]
Closing this as the issue is now gone since updating immich to v1.118.1
